### PR TITLE
Add instructions to setup PostgreSQL

### LINKED ISSUE #229: Database configuration guide
There should be a guide on database configuration explaining where to put the database link and what it should look like with majorly used databases (sqlite, mysql, postgresql)

### PR DESCRIPTION
**Pull request by mzelenyuk**
When generated a new Hanami project with PostgreSQL I was a bit confused how to setup a database on my local machine (I was not able to start `hanami server` because of database connection error). It's not well documented yet so I spend some time to find the answer.
I'm not sure where is the best place to show the setup instructions but I hope it will be helpful for new Hanami users.
Particularly closes #229.
Cheers!
postgresql setup

### CODE REVIEW

**Comment by TiteiKo** [comment]
Hi @mzelenyuk and thank you for the time spent on this!

As I said in the issue, I think it should be a full guide page, detailing the configuration for the 3 mainly used DBs. The Getting Started guide should stay clear of this, and only have a link to the previously mentioned Guide.

Do you think you'd have time to make a new guide page and adapt what you wrote here as a section of that guide?

**Comment by mzelenyuk** [comment]
Hi @TiteiKo! Yeah, I think a new Guide page for database configuration is the best case.
And sure, I'm able to do it if you'll help me with its title (I assume the "Database configuration" will be fine, what do you think?) and placement in right navigation list.

**Comment by TiteiKo** [comment]
I'm not the best to answer this, so I'll let the amazing people of @hanami/documentation answer your questions!

**Comment by mzelenyuk** [comment]
@TiteiKo Great, thanks!

**Comment by davydovanton** [comment]
@mzelenyuk good idea 👍 
But I think we need to create this section is more general (not only for psql DB).

What do you think, maybe will better to write something like this?

```
Before starting your server, you need to configure DB link in .env* files.
Open this file for each environment and update DB_URL for your DB.
For example, for psql, you need to do this and this ...
```

/cc @cllns

**Comment by mzelenyuk** [comment]
@davydovanton Sounds good. We may consider to place each DB's specific content under its own header (see screenshot) and tell how to create a table for each environment to make database setup more easier for the beginners.

db config

**Comment by davydovanton** [comment]
@mzelenyuk awesome 👍

**Comment by mzelenyuk** [comment]
@davydovanton Thanks! So what's the plan now? Waiting for @cllns response?

**Comment by davydovanton** [comment]
yep :)

**Comment by jodosha** [comment]
@mzelenyuk Hi and thanks for this PR.

---

**TL;DR:** We can merge this PR to help new people confused by Postgres complexity, but in the long run we should use Sqlite for the getting started guide.

---

The less details we present in the getting started guide, the less we overwhelm newcomers.

While having an env vars guide is desirable as reference, we should remove env vars from the getting started guide.

To solve the problem that @mzelenyuk ran into, we should abandon Postgres in the getting started guide in favor of Sqlite. There are a few advantages:

  1. Sqlite is the default database when we generate a new project (`hanami new bookshelf`)
  2. Postgres is great, but we don't use its powerful features in the getting started guide
  3. Sqlite doesn't require developer's attention to fiddle with env vars and check if a server is running.

**Comment by TiteiKo** [comment]
@jodosha The getting started guide already uses sqlite, see https://github.com/hanami/hanami.github.io/pull/238

**Comment by mzelenyuk** [comment]
Done (see attached screenshots). By now it has a section only for PostgreSQL. We can add more sections later.
Also before merge may anyone check spelling one more time please? :)

database configuration page
navigation list

File: source/guides/database-configuration/overview.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+title: Guides - Database Configuration Overview
+---
+
+# Overview
+
+Before starting your server, you need to configure database link in <code>.env*</code> files.
```
Comment (TiteiKo) [inline source/guides/database-configuration/overview.md]: configure **the** database link

File: source/guides/database-configuration/overview.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+Setup database variable for the test environment:
+
+```
+# .env.test
+DATABASE_URL="postgres://username:password@localhost/bookshelf_test"
+```
+
+## Prepare tables in database
+
+After your database variables setup is done you need to create a table in your database before you're be able to launch a development server.
```
Comment (TiteiKo) [inline source/guides/database-configuration/overview.md]: you need to create the **database and run the migrations before being able** to launch a development server.

File: source/guides/database-configuration/overview.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+In your terminal, enter:
+
+```
+% bundle exec hanami db prepare
+```
+
+To create a table in database for the test environment, enter:
```
Comment (TiteiKo) [inline source/guides/database-configuration/overview.md]: To **setup your test environment database**, enter:

File: source/guides/database-configuration/overview.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+Setup database variable for the test environment:
+
+```
+# .env.test
+DATABASE_URL="postgres://username:password@localhost/bookshelf_test"
+```
+
+## Prepare tables in database
```
Comment (TiteiKo) [inline source/guides/database-configuration/overview.md]: This could be at the same level of title as `Overview`

Comment (TiteiKo) [inline source/guides/database-configuration/overview.md]: Setup your database

**Comment by mzelenyuk** [comment]
@TiteiKo Fixed. Thanks for help! :)

**Comment by cllns** [comment]
@mzelenyuk Can you rename this `Models -> Database Configuration` (or something like this, if that's too long for the layout) instead of `Database Configuration -> Overview`? 

DB config only affects Models so I think it fits well in that group :)

**Comment by mzelenyuk** [comment]
@cllns Yes, sure.

updated navigation list

**Comment by jodosha** [comment]
@TiteiKo 
> The getting started guide already uses sqlite, see #238

Wonderful! 💯

**Comment by jodosha** [comment]
@cllns @TiteiKo If you think this is ready, please merge it 😉

**Comment by TiteiKo** [comment]
@cllns Good for me

File: source/guides/models/database-configuration.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+Open this file for each environment and update <code>DATABASE_URL</code> for your database.
+
+## <a href="http://www.postgresql.org/" target="_blank">PostgreSQL</a>
+
+Setup database variable for the development environment:
+
+```
+# .env.development
+DATABASE_URL="postgres://username:password@localhost/bookshelf_development"
```
Comment (AlfonsoUceda) [inline source/guides/models/database-configuration.md]: @mzelenyuk could you change `postgres://` to `postgresql://`? We are generating `postgresql://` because it supporter in MRI and JRuby ;)

File: source/guides/models/database-configuration.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+Setup database variable for the test environment:
+
+```
+# .env.test
+DATABASE_URL="postgres://username:password@localhost/bookshelf_test"
```
Comment (AlfonsoUceda) [inline source/guides/models/database-configuration.md]: the same here.

**Comment by mzelenyuk** [comment]
@AlfonsoUceda Yes, sure! :)

**Comment by AlfonsoUceda** [comment]
Thanks @mzelenyuk ;)
